packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from "vscode";
+
+const ControlCodes = {
+  CtrlC: "\u0003",
+};
+
+export abstract class BaseTaskTerminal implements vscode.Pseudoterminal {
+  protected writeEmitter = new vscode.EventEmitter<string>();
+  onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+  protected closeEmitter = new vscode.EventEmitter<number>();
+  onDidClose?: vscode.Event<number> = this.closeEmitter.event;
+
+  constructor(private taskDefinition: vscode.TaskDefinition) {}
+
+  open(): void {
+    this.do()
+      .then(() => this.stop())
```

Comment:
Does this style applies to background task, i.e., starting tunnel?

Reply:
Yes, this will apply to the background task. If the background task is running, the do func will not be resolved.

Reply:
So how to stop the background task when stopping debugging?

Reply:
> So how to stop the background task managed by custom terminal when stopping debugging?

close() will be called,